### PR TITLE
Add another error code for retry of Cosmos DB service on the timeout transient faults

### PR DESCRIPTION
**Pull request by rmt2021**
## Description
Add 408 (RequestTimeout) besides 503 to trigger retries for the Cosmos DB when transient faults happen.

## Related issues
Addresses https://github.com/microsoft/fhir-server/issues/2732. Also related to https://github.com/microsoft/fhir-server/pull/2106 which adds retries for 503 timeout transient error.

### CODE REVIEW

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/RetryExceptionPolicyFactory.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ private static AsyncRetryPolicy CreateExtendedRetryPolicy(int maxRetries, int ma
         {
             return Policy.Handle<RequestRateExceededException>()
                 .Or<CosmosException>(e => e.IsRequestRateExceeded())
-                .Or<CosmosException>(e => e.StatusCode == System.Net.HttpStatusCode.ServiceUnavailable)
+                .Or<CosmosException>(e => (e.StatusCode == System.Net.HttpStatusCode.ServiceUnavailable || e.StatusCode == System.Net.HttpStatusCode.RequestTimeout))
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/RetryExceptionPolicyFactory.cs:61]: Could we add a unit test for this? It looks like the 503 is being tested in CosmosFhirDataStoreTests.cs, line 204.

**Comment by LTA-Thinking** [comment]
@rmt2021 Thank you for your contribution to the repository. I wanted to check in and see if this PR was still on your radar.

**Comment by rmt2021** [comment]
@LTA-Thinking Yes! Sorry for the delay. I will try to write an unit test and let you review.
Thanks!

**Pull request merged**